### PR TITLE
Update `deepspeed` and `fairscale` versions

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Part of #12373.

- Updates deepspeed version used in CI from `0.5.7` to `0.5.10 (as of now)` and avoids using `0.6.{0,1}` due to https://github.com/microsoft/DeepSpeed/issues/1878
- Updates fairscale version used in CI from `0.4.0` to `0.4.6 (as of now)`
- Adapts deepspeed tests to `>=0.5.9`. The change in DeepSpeed API was introduced in https://github.com/microsoft/DeepSpeed/pull/1607.

### Does your PR introduce any breaking changes? If yes, please list them.
None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda @SeanNaren @awaelchli @rohitgr7

### CODE REVIEW

File: .azure-pipelines/gpu-tests.yml [hunk 1]
```diff
@@ -53,7 +53,7 @@ jobs:
     - bash: |
         python -c "fname = 'requirements/extra.txt' ; lines = [line for line in open(fname).readlines() if 'horovod' not in line] ; open(fname, 'w').writelines(lines)"
         pip install fairscale>=0.4.5
```
Comment (Borda) [inline .azure-pipelines/gpu-tests.yml:55]:
```suggestion
        pip install "fairscale>=0.4.5"
```

File: setup.cfg [hunk 2]
```diff
@@ -24,6 +24,7 @@ addopts =
     --doctest-modules
     --color=yes
     --disable-pytest-warnings
+    --ignore=tests/utilities/test_warnings.py
```
Comment (awaelchli) [inline setup.cfg:27]: why is this needed?

**Review by awaelchli — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

File: setup.cfg (hunk 2 above)
Comment (Borda) [reply setup.cfg:27]: @akihironitta ^^

Comment (akihironitta) [reply setup.cfg]: Reverted!

**Comment by akihironitta** [comment]
@carmocca @Borda Addressed @awaelchli's comment. Could you have another look?

**Review by Borda — APPROVED** [review]

File: .azure-pipelines/gpu-tests.yml [hunk 3]
```diff
@@ -52,8 +52,8 @@ jobs:
 
     - bash: |
         python -c "fname = 'requirements/extra.txt' ; lines = [line for line in open(fname).readlines() if 'horovod' not in line] ; open(fname, 'w').writelines(lines)"
-        pip install fairscale>=0.4.5
-        pip install deepspeed>=0.6.0
+        pip install "fairscale>=0.4.5"
```
[the remark this answers is not in the document]
Comment (Borda) [reply .azure-pipelines/gpu-tests.yml:55]: yep, I know, it just creates a file `0.4.5` :D
